packages/runtime-core/src/hydrationStrategies.ts

```diff
@@ -1,6 +1,28 @@
 import { isString } from '@vue/shared'
 import { DOMNodeTypes, isComment } from './hydration'
 
+// https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/compat/idle-callback.ts
+// Polyfills for Safari support
+// https://caniuse.com/requestidlecallback
+const requestIdleCallback: Window['requestIdleCallback'] =
+  globalThis.requestIdleCallback ||
+  (cb => {
+    const start = Date.now()
+    const idleDeadline = {
+      didTimeout: false,
+      timeRemaining: () => Math.max(0, 50 - (Date.now() - start)),
+    }
+    return setTimeout(() => {
+      cb(idleDeadline)
```

Comment:
The hydrate function doesn't make use of the deadline object, so it is not necessary. The polyfill could just be `setTimeout(cb, 1)`.